x-pack/solutions/security/plugins/elastic_assistant/server/lib/attack_discovery/schedules/fields/field_map.ts

```diff
@@ -0,0 +1,123 @@
+import { FieldMap, alertFieldMap } from '@kbn/alerts-as-data-utils';
+import {
+  ALERT_ATTACK_DISCOVERY_ALERTS_CONTEXT_COUNT,
+  ALERT_ATTACK_DISCOVERY_ALERT_IDS,
+  ALERT_ATTACK_DISCOVERY_API_CONFIG,
+  ALERT_ATTACK_DISCOVERY_API_CONFIG_ACTION_TYPE_ID,
+  ALERT_ATTACK_DISCOVERY_API_CONFIG_CONNECTOR_ID,
+  ALERT_ATTACK_DISCOVERY_API_CONFIG_DEFAULT_SYSTEM_PROMPT_ID,
+  ALERT_ATTACK_DISCOVERY_API_CONFIG_MODEL,
+  ALERT_ATTACK_DISCOVERY_API_CONFIG_PROVIDER,
+  ALERT_ATTACK_DISCOVERY_DETAILS_MARKDOWN,
+  ALERT_ATTACK_DISCOVERY_ENTITY_SUMMARY_MARKDOWN,
+  ALERT_ATTACK_DISCOVERY_MITRE_ATTACK_TACTICS,
+  ALERT_ATTACK_DISCOVERY_REPLACEMENTS,
+  ALERT_ATTACK_DISCOVERY_SUMMARY_MARKDOWN,
+  ALERT_ATTACK_DISCOVERY_TITLE,
+  ALERT_ATTACK_DISCOVERY_USERS,
+  ALERT_ATTACK_DISCOVERY_USER_ID,
+  ALERT_ATTACK_DISCOVERY_USER_NAME,
+} from './field_names';
+
+export const attackDiscoveryAlertFieldMap: FieldMap = {
+  /**
+   * Default alert-as-data fields
+   */
+  ...alertFieldMap,
+
+  /**
+   * Attack discovery fields
+   */
+  [ALERT_ATTACK_DISCOVERY_USERS]: {
+    type: 'nested',
+    array: true,
+    required: false,
+  },
+  [ALERT_ATTACK_DISCOVERY_USER_ID]: {
+    type: 'keyword',
+    array: false,
+    required: true,
+  },
+  [ALERT_ATTACK_DISCOVERY_USER_NAME]: {
+    type: 'keyword',
+    array: false,
+    required: false,
+  },
+  [ALERT_ATTACK_DISCOVERY_DETAILS_MARKDOWN]: {
+    type: 'text',
+    array: false,
+    required: true,
+  },
+  [ALERT_ATTACK_DISCOVERY_TITLE]: {
+    type: 'text',
+    array: false,
+    required: true,
+  },
+  [ALERT_ATTACK_DISCOVERY_ENTITY_SUMMARY_MARKDOWN]: {
+    type: 'text',
+    array: false,
+    required: true,
+  },
+  [ALERT_ATTACK_DISCOVERY_SUMMARY_MARKDOWN]: {
+    type: 'text',
+    array: false,
+    required: true,
+  },
+  [ALERT_ATTACK_DISCOVERY_MITRE_ATTACK_TACTICS]: {
+    type: 'keyword',
+    array: true,
+    required: false,
+  },
+  [ALERT_ATTACK_DISCOVERY_ALERT_IDS]: {
+    type: 'keyword',
+    array: true,
+    required: true,
+  },
+  [ALERT_ATTACK_DISCOVERY_REPLACEMENTS]: {
+    type: 'object',
+    array: false,
+    required: false,
+  },
+  [ALERT_ATTACK_DISCOVERY_API_CONFIG]: {
+    type: 'object',
+    array: false,
+    required: true,
+  },
+  [ALERT_ATTACK_DISCOVERY_API_CONFIG_CONNECTOR_ID]: {
+    type: 'keyword',
+    array: false,
+    required: true,
+  },
+  [ALERT_ATTACK_DISCOVERY_API_CONFIG_ACTION_TYPE_ID]: {
+    type: 'keyword',
+    array: false,
+    required: false,
+  },
+  [ALERT_ATTACK_DISCOVERY_API_CONFIG_DEFAULT_SYSTEM_PROMPT_ID]: {
+    type: 'keyword',
+    array: false,
+    required: false,
+  },
+  [ALERT_ATTACK_DISCOVERY_API_CONFIG_PROVIDER]: {
+    type: 'keyword',
+    array: false,
+    required: false,
+  },
+  [ALERT_ATTACK_DISCOVERY_API_CONFIG_MODEL]: {
+    type: 'keyword',
+    array: false,
+    required: false,
+  },
+  [ALERT_ATTACK_DISCOVERY_ALERTS_CONTEXT_COUNT]: {
+    type: 'integer',
+    array: false,
+    required: false,
+  },
+} as const;
```

Comment:
Consider adding a required `GENERATION_UUID` field to uniquely identify the run that generated the attack discovery (for ad hoc runs).

For scheduled runs, this field would have the same value as `kibana.alert.rule.execution.uuid`.

Reply:
Right, we need one! I was thinking if we could reuse existing `kibana.alert.rule.execution.uuid` field? It is already part of `alertFieldMap`. Also, there is `kibana.alert.rule.execution.type` field that meant to indicate manual vs scheduled rule run (`ALERT_RULE_EXECUTION_TYPE`) which we might want to set on alert creation.

Reply:
> Right, we need one! I was thinking if we could reuse existing `kibana.alert.rule.execution.uuid` field? It is already part of `alertFieldMap`. 

I would rather have one field than two, so if it's possible to write ad hoc alerts with an `kibana.alert.rule.execution.uuid` that doesn't correspond to a "real" execution, that sounds OK to me.

@mikecote do you have any concerns with the `kibana.alert.rule.execution.uuid` field being populated with a UUID that doesn't correlate to a real rule execution run?

> Also, there is `kibana.alert.rule.execution.type` field that meant to indicate manual vs scheduled rule run (`ALERT_RULE_EXECUTION_TYPE`) which we might want to set on alert creation.

That sounds good.